uportal-war/src/main/java/org/jasig/portal/portlets/search/SearchPortletController.java

```diff
@@ -719,12 +723,18 @@ private void addSearchResults(SearchResults portletSearchResults, PortalSearchRe
      */
     protected void modifySearchResultLinkTitle(SearchResult result, final HttpServletRequest httpServletRequest,
                                                final IPortletWindowId portletWindowId) {
-        if (result.getType().size() > 0 && searchResultsLinkTextPortletTitleSubstitutionSet.contains(result.getType().get(0))) {
-            final IPortletWindow portletWindow = this.portletWindowRegistry.getPortletWindow(httpServletRequest, portletWindowId);
+        // If the title contains a SpEL expression, parse it with the portlet definition in the evaluation context.
+        if (result.getType().size() > 0 && result.getTitle().contains("${")) {
+            final IPortletWindow portletWindow = this.portletWindowRegistry.getPortletWindow(httpServletRequest,
+                    portletWindowId);
             final IPortletEntity portletEntity = portletWindow.getPortletEntity();
             final IPortletDefinition portletDefinition = portletEntity.getPortletDefinition();
-            String portletTitle = portletDefinition.getTitle();
-            result.setTitle(portletTitle);
+            final SpELEnvironmentRoot spelEnvironment = new SpELEnvironmentRoot(portletDefinition);
+            try {
+                result.setTitle(spELService.getValue(result.getTitle(), spelEnvironment));
+            } catch (SpelParseException e) {
+                result.setTitle("Invalid Spring EL expression '" + result.getTitle()+"'");
```

Comment:
I had considered that.  I thought it might be more helpful for the admin to easily see the error and not have to figure out he has to dig into a stacktrace to figure out what happened.  Since the typical case is that the value is from a portlet.xml preference and is not likely to be set it seemed reasonable to me at the time.